server/client.go

```diff
@@ -1010,8 +1010,19 @@ func (c *client) traceMsg(msg []byte) {
 	if !c.trace {
 		return
 	}
-	// FIXME(dlc), allow limits to printable payload.
-	c.Tracef("<<- MSG_PAYLOAD: [%q]", msg[:len(msg)-LEN_CR_LF])
+
+	origLength := len(msg) - LEN_CR_LF
+	validLength := origLength
+	if origLength > MAX_TRACED_MESSAGE_PRINTABLE_LEN {
+		validLength = MAX_TRACED_MESSAGE_PRINTABLE_LEN
+		if validLength >= 3 {
+			msg[validLength-3] = '.'
```

Comment:
Should not write over msg, should just do a slice on original and then add '...'.